api/api.go

```diff
@@ -1,35 +1,54 @@
 package api
 
+//go:generate moq -out mockauth_test.go . AuthHandler
+
 import (
 	"context"
 	"net/http"
 
+	"github.com/ONSdigital/dp-authorisation/auth"
 	"github.com/ONSdigital/dp-healthcheck/healthcheck"
 	"github.com/ONSdigital/dp-recipe-api/config"
 	"github.com/ONSdigital/dp-recipe-api/store"
+	"github.com/ONSdigital/go-ns/identity"
 	"github.com/ONSdigital/go-ns/server"
 	"github.com/ONSdigital/log.go/log"
 	"github.com/gorilla/mux"
 	"github.com/justinas/alice"
 )
 
-var srv *server.Server
+var (
+	srv *server.Server
+
+	create = auth.Permissions{Create: true}
+	update = auth.Permissions{Update: true}
+)
+
+// AuthHandler provides authorisation checks on requests
+type AuthHandler interface {
+	Require(required auth.Permissions, handler http.HandlerFunc) http.HandlerFunc
+}
 
 //RecipeAPI contains store and features for managing the recipe
 type RecipeAPI struct {
-	dataStore        store.DataStore
-	Router           *mux.Router
-	EnableAuthImport bool
+	dataStore              store.DataStore
+	Router                 *mux.Router
+	recipePermissions      AuthHandler
+	permissions            AuthHandler
+	EnablePrivateEndpoints bool
 }
 
 //CreateAndInitialiseRecipeAPI create a new RecipeAPI instance based on the configuration provided and starts the HTTP server.
-func CreateAndInitialiseRecipeAPI(ctx context.Context, cfg config.Configuration, dataStore store.DataStore, hc *healthcheck.HealthCheck, errorChan chan error) {
+func CreateAndInitialiseRecipeAPI(ctx context.Context, cfg config.Configuration, dataStore store.DataStore, hc *healthcheck.HealthCheck, errorChan chan error, recipePermissions AuthHandler, permissions AuthHandler) {
 	router := mux.NewRouter()
-	api := NewRecipeAPI(ctx, cfg, router, dataStore)
+	api := NewRecipeAPI(ctx, cfg, router, dataStore, recipePermissions, permissions)
 
 	healthcheckHandler := newMiddleware(hc.Handler)
 	middleware := alice.New(healthcheckHandler)
 
+	// Add the identity middleware when running in publishing
```

Comment:
This always gets run, whether in publishing or web?

Reply:
This removed in commit "remove identity middleware to avoid authenticating whole API"